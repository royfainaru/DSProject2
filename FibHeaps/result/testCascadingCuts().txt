╮
├── ╭────────────────╮
│   │ Key: -1        │
│   ├────────────────┤
│   │ Rank: 1        │
│   │ Marked: false  │
│   │ Parent: (null) │
│   │ Next: 2        │
│   │ Prev: (null)   │
│   │ Child: 17      │
│   ╰────────────────╯
│   ╰── ╭───────────────╮
│       │ Key: 17       │
│       ├───────────────┤
│       │ Rank: 0       │
│       │ Marked: false │
│       │ Parent: -1    │
│       │ Next: (null)  │
│       │ Prev: (null)  │
│       │ Child: (null) │
│       ╰───────────────╯
╰── ╭────────────────╮
    │ Key: 2         │
    ├────────────────┤
    │ Rank: 4        │
    │ Marked: false  │
    │ Parent: (null) │
    │ Next: (null)   │
    │ Prev: -1       │
    │ Child: 10      │
    ╰────────────────╯
    ├── ╭──────────────╮
    │   │ Key: 10      │
    │   ├──────────────┤
    │   │ Rank: 3      │
    │   │ Marked: true │
    │   │ Parent: 2    │
    │   │ Next: 5      │
    │   │ Prev: (null) │
    │   │ Child: 14    │
    │   ╰──────────────╯
    │   ├── ╭──────────────╮
    │   │   │ Key: 14      │
    │   │   ├──────────────┤
    │   │   │ Rank: 1      │
    │   │   │ Marked: true │
    │   │   │ Parent: 10   │
    │   │   │ Next: 12     │
    │   │   │ Prev: (null) │
    │   │   │ Child: 15    │
    │   │   ╰──────────────╯
    │   │   ╰── ╭───────────────╮
    │   │       │ Key: 15       │
    │   │       ├───────────────┤
    │   │       │ Rank: 0       │
    │   │       │ Marked: false │
    │   │       │ Parent: 14    │
    │   │       │ Next: (null)  │
    │   │       │ Prev: (null)  │
    │   │       │ Child: (null) │
    │   │       ╰───────────────╯
    │   ├── ╭───────────────╮
    │   │   │ Key: 12       │
    │   │   ├───────────────┤
    │   │   │ Rank: 1       │
    │   │   │ Marked: false │
    │   │   │ Parent: 10    │
    │   │   │ Next: 11      │
    │   │   │ Prev: 14      │
    │   │   │ Child: 13     │
    │   │   ╰───────────────╯
    │   │   ╰── ╭───────────────╮
    │   │       │ Key: 13       │
    │   │       ├───────────────┤
    │   │       │ Rank: 0       │
    │   │       │ Marked: false │
    │   │       │ Parent: 12    │
    │   │       │ Next: (null)  │
    │   │       │ Prev: (null)  │
    │   │       │ Child: (null) │
    │   │       ╰───────────────╯
    │   ╰── ╭───────────────╮
    │       │ Key: 11       │
    │       ├───────────────┤
    │       │ Rank: 0       │
    │       │ Marked: false │
    │       │ Parent: 10    │
    │       │ Next: (null)  │
    │       │ Prev: 12      │
    │       │ Child: (null) │
    │       ╰───────────────╯
    ├── ╭───────────────╮
    │   │ Key: 5        │
    │   ├───────────────┤
    │   │ Rank: 2       │
    │   │ Marked: false │
    │   │ Parent: 2     │
    │   │ Next: 4       │
    │   │ Prev: 10      │
    │   │ Child: 6      │
    │   ╰───────────────╯
    │   ├── ╭───────────────╮
    │   │   │ Key: 6        │
    │   │   ├───────────────┤
    │   │   │ Rank: 1       │
    │   │   │ Marked: false │
    │   │   │ Parent: 5     │
    │   │   │ Next: 9       │
    │   │   │ Prev: (null)  │
    │   │   │ Child: 8      │
    │   │   ╰───────────────╯
    │   │   ╰── ╭───────────────╮
    │   │       │ Key: 8        │
    │   │       ├───────────────┤
    │   │       │ Rank: 0       │
    │   │       │ Marked: false │
    │   │       │ Parent: 6     │
    │   │       │ Next: (null)  │
    │   │       │ Prev: (null)  │
    │   │       │ Child: (null) │
    │   │       ╰───────────────╯
    │   ╰── ╭───────────────╮
    │       │ Key: 9        │
    │       ├───────────────┤
    │       │ Rank: 0       │
    │       │ Marked: false │
    │       │ Parent: 5     │
    │       │ Next: (null)  │
    │       │ Prev: 6       │
    │       │ Child: (null) │
    │       ╰───────────────╯
    ├── ╭───────────────╮
    │   │ Key: 4        │
    │   ├───────────────┤
    │   │ Rank: 1       │
    │   │ Marked: false │
    │   │ Parent: 2     │
    │   │ Next: 3       │
    │   │ Prev: 5       │
    │   │ Child: 7      │
    │   ╰───────────────╯
    │   ╰── ╭───────────────╮
    │       │ Key: 7        │
    │       ├───────────────┤
    │       │ Rank: 0       │
    │       │ Marked: false │
    │       │ Parent: 4     │
    │       │ Next: (null)  │
    │       │ Prev: (null)  │
    │       │ Child: (null) │
    │       ╰───────────────╯
    ╰── ╭───────────────╮
        │ Key: 3        │
        ├───────────────┤
        │ Rank: 0       │
        │ Marked: false │
        │ Parent: 2     │
        │ Next: (null)  │
        │ Prev: 4       │
        │ Child: (null) │
        ╰───────────────╯
