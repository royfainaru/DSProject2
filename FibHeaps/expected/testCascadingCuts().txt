╮
├── ╭────────────────╮
│   │ Key: 14        │
│   ├────────────────┤
│   │ Rank: 0        │
│   │ Marked: false  │
│   │ Parent: (null) │
│   │ Next: -1       │
│   │ Prev: 2        │
│   │ Child: (null)  │
│   ╰────────────────╯
├── ╭────────────────╮
│   │ Key: -1        │
│   ├────────────────┤
│   │ Rank: 1        │
│   │ Marked: false  │
│   │ Parent: (null) │
│   │ Next: -2       │
│   │ Prev: 14       │
│   │ Child: 17      │
│   ╰────────────────╯
│   ╰── ╭───────────────╮
│       │ Key: 17       │
│       ├───────────────┤
│       │ Rank: 0       │
│       │ Marked: false │
│       │ Parent: -1    │
│       │ Next: 17      │
│       │ Prev: 17      │
│       │ Child: (null) │
│       ╰───────────────╯
├── ╭────────────────╮
│   │ Key: -2        │
│   ├────────────────┤
│   │ Rank: 2        │
│   │ Marked: false  │
│   │ Parent: (null) │
│   │ Next: 2        │
│   │ Prev: -1       │
│   │ Child: 10      │
│   ╰────────────────╯
│   ├── ╭───────────────╮
│   │   │ Key: 10       │
│   │   ├───────────────┤
│   │   │ Rank: 1       │
│   │   │ Marked: false │
│   │   │ Parent: -2    │
│   │   │ Next: 13      │
│   │   │ Prev: 13      │
│   │   │ Child: 11     │
│   │   ╰───────────────╯
│   │   ╰── ╭───────────────╮
│   │       │ Key: 11       │
│   │       ├───────────────┤
│   │       │ Rank: 0       │
│   │       │ Marked: false │
│   │       │ Parent: 10    │
│   │       │ Next: 11      │
│   │       │ Prev: 11      │
│   │       │ Child: (null) │
│   │       ╰───────────────╯
│   ╰── ╭───────────────╮
│       │ Key: 13       │
│       ├───────────────┤
│       │ Rank: 0       │
│       │ Marked: false │
│       │ Parent: -2    │
│       │ Next: 10      │
│       │ Prev: 10      │
│       │ Child: (null) │
│       ╰───────────────╯
╰── ╭────────────────╮
    │ Key: 2         │
    ├────────────────┤
    │ Rank: 3        │
    │ Marked: false  │
    │ Parent: (null) │
    │ Next: 14       │
    │ Prev: -2       │
    │ Child: 5       │
    ╰────────────────╯
    ├── ╭───────────────╮
    │   │ Key: 5        │
    │   ├───────────────┤
    │   │ Rank: 2       │
    │   │ Marked: false │
    │   │ Parent: 2     │
    │   │ Next: 4       │
    │   │ Prev: 3       │
    │   │ Child: 6      │
    │   ╰───────────────╯
    │   ├── ╭───────────────╮
    │   │   │ Key: 6        │
    │   │   ├───────────────┤
    │   │   │ Rank: 1       │
    │   │   │ Marked: false │
    │   │   │ Parent: 5     │
    │   │   │ Next: 9       │
    │   │   │ Prev: 9       │
    │   │   │ Child: 8      │
    │   │   ╰───────────────╯
    │   │   ╰── ╭───────────────╮
    │   │       │ Key: 8        │
    │   │       ├───────────────┤
    │   │       │ Rank: 0       │
    │   │       │ Marked: false │
    │   │       │ Parent: 6     │
    │   │       │ Next: 8       │
    │   │       │ Prev: 8       │
    │   │       │ Child: (null) │
    │   │       ╰───────────────╯
    │   ╰── ╭───────────────╮
    │       │ Key: 9        │
    │       ├───────────────┤
    │       │ Rank: 0       │
    │       │ Marked: false │
    │       │ Parent: 5     │
    │       │ Next: 6       │
    │       │ Prev: 6       │
    │       │ Child: (null) │
    │       ╰───────────────╯
    ├── ╭───────────────╮
    │   │ Key: 4        │
    │   ├───────────────┤
    │   │ Rank: 1       │
    │   │ Marked: false │
    │   │ Parent: 2     │
    │   │ Next: 3       │
    │   │ Prev: 5       │
    │   │ Child: 7      │
    │   ╰───────────────╯
    │   ╰── ╭───────────────╮
    │       │ Key: 7        │
    │       ├───────────────┤
    │       │ Rank: 0       │
    │       │ Marked: false │
    │       │ Parent: 4     │
    │       │ Next: 7       │
    │       │ Prev: 7       │
    │       │ Child: (null) │
    │       ╰───────────────╯
    ╰── ╭───────────────╮
        │ Key: 3        │
        ├───────────────┤
        │ Rank: 0       │
        │ Marked: false │
        │ Parent: 2     │
        │ Next: 5       │
        │ Prev: 4       │
        │ Child: (null) │
        ╰───────────────╯
